app/models.py

```diff
@@ -0,0 +1,230 @@
+from peewee import *
+
+DATABASE_NAME = "database.db"
+database = SqliteDatabase(DATABASE_NAME)
+
+
+# Database models
+class BaseModel(Model):
+    class Meta:
+        database = database
+
+
+class User(BaseModel):
+    username = CharField(unique=True)
+    fullname = CharField()
+    mail_address = CharField()
+    password = CharField()
+    user_type = CharField()
+
+    def is_student(self):
+        return User.user_type == "is_student"
+
+    def is_teacher(self):
+        return User.user_type == "teacher"
+
+    def get_user_courses(self):
+        return (Course
+                .select()
+                .join(UserCourseRelationship, on=UserCourseRelationship.course)
+                .where(UserCourseRelationship.user == self)
+                .order_by(Course.id)
+                )
+
+
+class Course(BaseModel):
+    course_id = IntegerField(unique=True)
+    name = CharField()
+    open_date = DateTimeField()
+
+    def get_course_administrators(self):
+        return (User
+                .select()
+                .join(UserCourseRelationship, on=UserCourseRelationship.user)
+                .where((UserCourseRelationship.course == self) & UserCourseRelationship.is_admin))
+
+
+class UserCourseRelationship(BaseModel):
+    user = ForeignKeyField(User, backref='user')
+    course = ForeignKeyField(Course, backref='course')
+    is_admin = BooleanField(default=False)
+
+
+class Lecture(BaseModel):
```

Comment:
What do you think about modifying this to have `start_time` and `end_time`?

Reply:
Sounds good!
In general, as I discussed with Yam, both Course and Lecture will possibly need more columns (Just few examples that come to mind: an end date for course, reference to lecture's materials, teacher's notes, a teacher user with lower permissions that an admin, and such).
Bearing that in mind, we will need to implement and create the database with caution, as we will possibly need to migrate them in the future.